R/databricks.R

```diff
@@ -0,0 +1,159 @@
+#' Helper for Connecting to Databricks via ODBC
+#'
+#' @description
+#'
+#' Connect to Databricks clusters and SQL warehouses via the [Databricks ODBC
+#' driver](https://www.databricks.com/spark/odbc-drivers-download).
+#'
+#' In particular, the custom `dbConnect()` method for the Databricks ODBC driver
+#' implements a subset of the [Databricks client unified authentication](https://docs.databricks.com/en/dev-tools/auth.html#databricks-client-unified-authentication)
+#' model, with support for personal access tokens, OAuth machine-to-machine
+#' credentials, and OAuth user-to-machine credentials supplied via Posit
+#' Workbench or the Databricks CLI on desktop.
+#'
+#' All of these credentials are detected automatically if present using
+#' [standard environment variables](https://docs.databricks.com/en/dev-tools/auth.html#environment-variables-and-fields-for-client-unified-authentication).
+#'
+#' @param http_path To query a cluster, use the HTTP Path value found under
+#'   `Advanced Options > JDBC/ODBC` in the Databricks UI. For SQL warehouses,
+#'   this is found under `Connection Details` instead.
+#' @param workspace The URL of a Databricks workspace, e.g.
+#'   `"https://example.cloud.databricks.com"`.
+#' @param driver The name of the Databricks ODBC driver, or `NULL` to use the
+#'   default name.
+#' @param ... Further arguments passed on to [`dbConnect()`].
+#'
+#' @returns An `OdbcConnection` object with an active connection to a Databricks
+#'   cluster or SQL warehouse.
+#'
+#' @examples
+#' \dontrun{
+#' DBI::dbConnect(
+#'   odbc::databricks(),
+#'   http_path = "sql/protocolv1/o/4425955464597947/1026-023828-vn51jugj"
+#' )
+#' }
+#' @export
+databricks <- function() {
+  new("DatabricksOdbcDriver")
+}
+
+#' @rdname databricks
+#' @export
+setClass("DatabricksOdbcDriver", contains = "OdbcDriver")
+
+#' @rdname databricks
+#' @export
+setMethod(
+  "dbConnect", "DatabricksOdbcDriver",
+  function(drv, http_path, workspace = Sys.getenv("DATABRICKS_HOST"),
+           driver = NULL, ...) {
+    args <- databricks_args(
+      http_path = http_path,
+      workspace = workspace,
+      driver = driver
+    )
+    args <- c(args, ...)
+    inject(callNextMethod(!!!args))
```

Comment:
I have zero trust for `callNextMethod()`, so I'd suggest something like:

```suggestion
    inject(dbConnect(odbc(), !!!args))
```